dask_sql/sql-schema.yaml

```diff
@@ -69,6 +69,38 @@ properties:
         description: |
           Whether to apply the dynamic partition pruning optimizer rule.
 
+      verbose_optimizer:
```

Comment:
I really don't like the name `verbose_optimizer`, but I couldn't really think of a better name...

Reply:
Yeah I'm no naming expert. I like the feature though.

Reply:
Generally would rather avoid this configuration for now, but if we do have it maybe it would make sense as something like `sql.explain.skipped_optimizers`? Then it could be a list where users pass any arbitrary rules, but it defaults to whatever rules we always want disabled for explains.